### PR TITLE
Redefined `ComplCaseSensitivity` as a proper sum type. (#509)

### PR DESCRIPTION
**Pull request by nikshalark**
### Description

This is my first time submitting a pull request of any kind, so please tell me if I've done anything incorrectly.

This PR is an implementation of the suggestion in issue #509, with all relevant functions changed so as to expect the rewritten datatype. 

### Checklist

  - [✔️] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [✔️] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [✔️] I updated the `CHANGES.md` file

  - [✔️] I updated the `XMonad.Doc.Extending` file (if appropriate)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -234,6 +234,10 @@
 
   * Add support for GHC 9.0.1.
 
+  * `XMonad.Prompt`
+
+    - Redefine `ComplCaseSensitivity` to a proper sum type as opposed to a newtype wrapped around Bool. This 
+
```
Comment (TheMC47) [inline CHANGES.md:240]: There is already an entry for `XMonad.Prompt`, so add it there instead.

**Review by TheMC47 — CHANGES_REQUESTED** [review]
Awesome, thank you for your contribution! :tada:  Just a few comments and we're ready to go

File: XMonad/Prompt/Shell.hs [hunk 2]
```diff
@@ -128,8 +128,8 @@ compgen csn actionOpt s = runProcessWithInput "bash" [] $
     complCaseSensitivityCmd csn ++ " ; " ++ compgenCmd actionOpt s
 
 complCaseSensitivityCmd :: ComplCaseSensitivity -> String
-complCaseSensitivityCmd (ComplCaseSensitive b) =
-    "bind 'set completion-ignore-case " ++ (if b then "off" else "on") ++ "'"
+complCaseSensitivityCmd b =
+    "bind 'set completion-ignore-case " ++ (if b == CaseSensitive then "off" else "on") ++ "'"
```
[the remark this answers is not in the document]
Comment (geekosaur) [reply XMonad/Prompt/Shell.hs:132]: Of the two, the first is more Haskelly. Eq costs more and is less idiomatic.

[the remark this answers is not in the document]
Comment (slotThe) [reply XMonad/Prompt/Shell.hs:132]: I agree, something like 

``` haskell
complCaseSensitivityCmd cs = [...]
 where
  caseSensitivity = case cs of 
    CaseSensitive   -> "off"
    CaseInsensitive -> "on"
```

would be nice!

**Comment by nikshalark** [comment]
Thank you everyone for being so kind in dealing with this, especially with such a small issue. I have implemented the changes you all have requested. Unnecessary parentheses have been removed and the equality issue has been solved with a pattern match.

File: XMonad/Prompt/Shell.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (nikshalark) [reply XMonad/Prompt/Shell.hs]: This has been changed! Thanks for catching this, I always seem to think an Eq instance is defined no matter what I'm working on.

**Comment by slotThe** [comment]
Thanks!

**Pull request merged**